.github/workflows/tests.yaml

```diff
@@ -0,0 +1,69 @@
+name: tests
+on: ["push", "pull_request", "workflow_dispatch"]
+
+jobs:
+  build-test:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go: ['1.18', '1.17', '1.16', '1.15', '1.14', '1.13']
+
+    name: basic-with-go-v${{ matrix.go }}"
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+
+      - name: Setup Go Version
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go }}
+
+      - name: Build
+        run: |
+          go build ./...
+
+      - name: Go Vet
+        run: |
+          go vet ./...
+
+      - name: Go Fmt
+        run: |
+          test "0" = $(gofmt -d . | wc -l)
+
+      - name: Go Test
+        run: |
+          go test ./...
+
+  static-check:
+    name: "static-check-linter-2022.1"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+
+      - uses: dominikh/staticcheck-action@v1.2.0
```

Comment:
Out of curiosity, what static checks are included in this action?

Reply:
https://staticcheck.io/

Discussed in 1-1 meeting.